ingest/src/main/java/com/microsoft/azure/kusto/ingest/source/BlobSourceInfo.java

```diff
@@ -66,4 +64,26 @@ public Map<String, String> getTracingAttributes() {
         }
         return attributes;
     }
+
+    /*
+     * For internal usage, adding blobExactSize
+     */
+    public static BlobSourceInfo fromFile(String blobPath, FileSourceInfo fileSourceInfo, CompressionType sourceCompressionType, boolean gotCompressed) {
```

Comment:
The param on the outside is "shouldCompress", but here it's called "gotCompressed" - any reason?

Reply:
Client should care if it should compress
 . file naming should care only that it was compressed .